### PR TITLE
Add support for llvm.[sdiv|udiv] and the arith counterparts

### PR DESCRIPTION
**Pull request by tobiasgrosser**
For now, division by zero is returned as `poison`. As we gain support for proper side effects, we should turn this into immediate undefined behavior.

### CODE REVIEW

File: Test/Interpreter/LLVM/udiv.mlir [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// RUN: veir-interpret %s | filecheck %s
+
+"builtin.module"() ({
+  %lhs = "llvm.constant"() <{ "value" = 130 : i32 }> : () -> i32
+  %rhs = "llvm.constant"() <{ "value" = 3 : i32 }> : () -> i32
+  %x = "llvm.mul"(%lhs, %rhs) : (i32, i32) -> i32
```
Comment (math-fehr) [inline Test/Interpreter/LLVM/udiv.mlir]: This should be an `udiv`, not a `mul`

File: Test/Interpreter/LLVM/udiv_exact.mlir [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// RUN: veir-interpret %s | filecheck %s
+
+"builtin.module"() ({
+  %lhs = "llvm.constant"() <{ "value" = 7 : i32 }> : () -> i32
+  %rhs = "llvm.constant"() <{ "value" = 2 : i32 }> : () -> i32
+  %x = "llvm.sdiv"(%lhs, %rhs) <{exact}> : (i32, i32) -> i32
```
Comment (math-fehr) [inline Test/Interpreter/LLVM/udiv_exact.mlir]: This should be an `udiv`, not an `sdiv`

**Review by math-fehr — COMMENTED** [review]
Looks good to me, I just thing we should need more tests:
* Fixing the current tests (which do not correspond to the file name)
* Adding tests for division by zero
* Adding tests for INT_MIN.sdiv -1 ?

File: Test/Interpreter/LLVM/udiv.mlir (hunk 1 above)
Comment (tobiasgrosser) [reply Test/Interpreter/LLVM/udiv.mlir]: Fixed.

File: Test/Interpreter/LLVM/udiv_exact.mlir (hunk 2 above)
Comment (tobiasgrosser) [reply Test/Interpreter/LLVM/udiv_exact.mlir]: Fixed.

**Comment by tobiasgrosser** [comment]
> Added. Currently, it seems negative constants are not parsed correctly. 😢

This has been resolved in #259. All test cases now pass.
